### PR TITLE
WIP update discovery to work with new multinic core

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Continuation of #143

### CODE REVIEW

**Comment by lzap** [comment]
Ok one minor failure - it's because on my branch (latest develop) the organization name is really not nil:

```
  1) Failure:
Api::V2::DiscoveredHostsControllerTest#test_show_host [/home/lzap/work/foreman_discovery/test/functional/api/v2/discovered_hosts_controller_test.rb:36]:
<nil> expected but was
<"Empty Organization">.
```

Fixture change, easy change.

Going to test that.

**Comment by lzap** [comment]
Ok testing the patch and getting

```
ActionView::Template::Error (undefined method `subnet' for nil:NilClass):
    34:       <% end -%>
    35:       <% if SETTINGS[:organizations_enabled] -%>
    36:         <td class="hidden-tablet hidden-xs"><%= host.organization.try(:name) %></td>
    37:       <% end -%>
    38:       <td class="hidden-tablet hidden-xs"><%= host.primary_interface.subnet %></td>
    39:       <td class="hidden-tablet hidden-xs"><%= disc_report_column(host) %></td>
    40:       <td>
  app/controllers/concerns/application_shared.rb:13:in `set_timezone'
  app/models/concerns/foreman/thread_session.rb:33:in `clear_thread'
  lib/middleware/catch_json_parse_errors.rb:9:in `call'
```

when I try to list discovered hosts in the interface. The host seems to be discovered correctly.

**Comment by lzap** [comment]
Wait it works, but the error message only appears when I have some old discovered hosts in the list. It looks like this might be issue during upgrade. We need to either workaround that, or ask users to delete all / reboot all discovered hosts.

File: app/views/discovered_hosts/_discovered_hosts_list.html.erb [hunk 1]
```diff
@@ -35,7 +35,7 @@
       <% if SETTINGS[:organizations_enabled] -%>
         <td class="hidden-tablet hidden-xs"><%= host.organization.try(:name) %></td>
       <% end -%>
-      <td class="hidden-tablet hidden-xs"><%= host.subnet %></td>
+      <td class="hidden-tablet hidden-xs"><%= host.primary_interface.subnet %></td>
```
Comment (lzap) [inline app/views/discovered_hosts/_discovered_hosts_list.html.erb]: We could do `host.primary_intarface.try(:subnet)` maybe or similar with a comment what this is for.

File: test/functional/api/v2/discovered_hosts_controller_test.rb [hunk 2]
```diff
@@ -31,7 +33,7 @@ def test_show_host
     assert_equal 42001, show_response["memory"]
     assert_equal 0, show_response["disk_count"]
     assert_equal 0, show_response["disks_size"]
-    assert_equal "Empty Organization", show_response["organization_name"]
+    assert_equal nil, show_response["organization_name"]
```
Comment (lzap) [inline test/functional/api/v2/discovered_hosts_controller_test.rb:36]: I still have "Empty Organization" there.

File: app/views/discovered_hosts/_discovered_hosts_list.html.erb (hunk 1 above)
Comment (lzap) [inline app/views/discovered_hosts/_discovered_hosts_list.html.erb]: It looks like V2 API / CLI does work well. No changes required there, this is the only place it looks like.

Comment (GregSutcliffe) [inline app/views/discovered_hosts/_discovered_hosts_list.html.erb]: Agreed, it's a nice safety catch. Added.

**Comment by GregSutcliffe** [comment]
@lzap tests are fine as they are, for me:

```
[greg:~/github/discovery_core]$ git log --oneline | head -n1
714efb3 WIP update discovery to work with new multinic core
[greg:~/github/foreman]$ git log --oneline | head -n1
796b718 Fixes #9147 - move managed checks to the start of the DHCP/DNS/TFTP ...
[greg:~/github/foreman]$  rm db/development.sqlite db/test.sqlite
[greg:~/github/foreman]$  rake db:migrate
[greg:~/github/foreman]$  rake db:test:prepare
[greg:~/github/foreman]$ rake test:discovery
Finished tests in 16.569241s, 4.5265 tests/s, 6.5181 assertions/s.
75 tests, 108 assertions, 0 failures, 0 errors, 2 skips
```

**Comment by domcleal** [comment]
http://ci.theforeman.org/job/test_plugin_pull_request/555/ is running, which may settle it?

Edit: oh, never mind.  I thought the core PR had been merged.

File: test/functional/api/v2/discovered_hosts_controller_test.rb (hunk 2 above)
Comment (lzap) [inline test/functional/api/v2/discovered_hosts_controller_test.rb:36]: Found the issue with this test and issued new PR with updated test.

**Pull request closed without merging**